src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReflectionEmitMemberAccessor.cs

```diff
@@ -221,33 +221,29 @@ private static DynamicMethod CreateImmutableDictionaryCreateRangeDelegate(Type c
         }
 
         public override Func<object?, TProperty> CreatePropertyGetter<TProperty>(PropertyInfo propertyInfo) =>
-            CreateDelegate<Func<object?, TProperty>>(CreatePropertyGetter(propertyInfo, propertyInfo.DeclaringType!, typeof(TProperty)));
+            CreateDelegate<Func<object?, TProperty>>(CreatePropertyGetter(propertyInfo));
```

Comment:
Looks like there are some CI failures relating to the delegate creation. I'll let you address those, then I'll circle back to the member accessor changes.

Reply:
Not sure that the failures are caused by this changes, because the exceptions says about the parameter mismatch. The problem comes from broken JSON property info creation I assume.

Reply:
Yep, this causes problems too. For example, `TestClassWithFieldsHavingCustomConverter` has property `DerivedCustomer` of type `DerivedCustomer`. The member accessor correctly generates methods using the `DerivedCustomer`, but a callee requested method for type `Customer`.

How this happens? `JsonConverter<T>` is responsible for a property creation for now, but its `T` can be less derived than the actual property type.

Reply:
Are there still issues with this?